### DISCUSSION TITLE
How to output renderer log to custom filePath

### DISCUSSION THREAD

**Discussion by heelaine** [megahertz/electron-log#411] [General]
i suppose to output the log in the renderer to a custom file, attempting to use IPC channel, but the log is being output to the main.log file. How can I customize the log file path in the rendering process? 
// main.ts 
import log from 'electron/main'
log.initialize({spyRenderedConsole:true})
log.transports.file.resolvePathFn = ()=>{ return join(path,'main.log')}

// renderer.ts
ipcRenderer.send('__ELECTRON_LOG__',{
    data: 'test',
    level:'info',
})

if i create logger instances,how to assign renderer instances to output log file path？

**Comment by megahertz** [comment]
You can define your path based on the second argument of  [`resolvePathFn`](https://github.com/megahertz/electron-log/blob/master/docs/transports/file.md#resolvepathfn-variables-pathvariables-message-logmessage--string)

check `msg.variables.processType`

**Reply by heelaine** [reply]
Sorry I didn't make myself clear. I've already configured the path of the logger file linked to the renderer. But the ipc renderer generates log in main.log by default. I was wondering how to change the default file location，msg.variables.processtype = 'renderer'.

**Reply by heelaine** [reply]
I'm still a beginner. Please pardon my mistakes....

**Reply by megahertz** [reply]
1. When you send  the `__ELECTRON_LOG__` event, you should manually add `variables: { processType: 'renderer' }` value there. I just updated the initialization sample in docs to include this value.

2. Once it's correctly set, you need to return a different path based on `msg.variables.processType`, like:

 ```js
 log.transports.file.resolvePathFn = (variables, msg) => {
   console.log({ variables, msg });
   if (msg?.variables.processType === 'renderer') {
     return 'someRendererPath';
   } else {
     return 'someMainPath';
   }
 };


 ```

**Reply by heelaine** [reply]
Much thanks. I solved the problem by following your instructions. Initially I was trying to control the file location by passing parameters. Later i found out that the files generated by the renderer include the logs of the main process. This was a result of my previous code, which was wrong
```
// main.ts
import log from 'electron/main'
function initLogger(path,logName){
     const logger = log.create({logId:logName})
     log.transports.file.resolvePathFn = (variables) => {
      return join(path,`logs/${logName}.log`)
    };
}
```
